lib/order.rb

```diff
@@ -1,18 +1,48 @@
 module Grocery
+  require 'csv'
   class Order
     attr_reader :id, :products
 
     def initialize(id, products)
-      @id = id
+      @id = id.to_i
       @products = products
     end
 
     def total
-      # TODO: implement total
+      sum = products.values.inject(0, :+)
+      return sum + (sum * 0.075).round(2)
     end
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      unless @products.has_key?(product_name) # as long as the collection of products doesn't already have the key of the new item, you can add it to the list
+        @products[product_name] = product_price
+        return true
+      else
+        return false # otherwise, it isn't unique, and something with that name is already on file.
+      end
+    end
+
+    def self.all
+      all_orders = []
+      CSV.open("support/orders.csv", 'r').each do |line|
+        id = line[0].to_i
+        products = {}
+        line[1].split(";").each do |item|
+          details = item.split(":")
+          products[details[0]] = details[1]
```

Comment:
You could change this loop to

```ruby
line[1].split(";").each do |item|
  name, price = item.split(":")
  products[name] = price
end
```

In my opinion that's a little more readable.